### PR TITLE
Add lifts to `Output[Option[List[A]]]` etc.

### LINKED ISSUE #424: Consider adding lifted methods for `Output[List[A]]` and `Output[Option[List[A]]]`, and `Output[Option[A]]`
In the following use case:
```
val nginxService = Service(
    "nginx",
    ServiceArgs(
      spec = ServiceSpecArgs(
        selector = labels,
        `type` = k8s.core.v1.enums.ServiceSpecType.LoadBalancer,
        ...
      ),...
  )
  
  Stack.exports(
    nginxUrl = nginxService.status.loadBalancer.ingress.head.hostname
  )
```

what is missing is:
```
  extension [A](output: besom.types.Output[scala.Option[scala.collection.immutable.List[A]]])
    def head: besom.types.Output[scala.Option[A]] = output.map(_.map(_.head))
```

We should consider adding lifts for both `Output[List[A]]` and `Output[Option[List[A]]]`, for the essential methods: i.e.: `head`, `tail`, `get`, etc. (`mapList`?), but adding every single method does not feel practical.

EDIT:

A more complex use case:
```
  extension [A](output: Output[Option[A]])
    def orElse[B >: A](alternative: Output[Option[B]]): Output[Option[B]] =
      output.flatMap(o => alternative.map(a => o.orElse(a)))

  extension [A](output: Output[Option[List[A]]]) 
    def headOption: Output[Option[A]] = output.map(_.flatMap(_.headOption))

  val maybeIngress = nginxService.status.loadBalancer.ingress.headOption
  val hostnameOrIp = maybeIngress.hostname.orElse(maybeIngress.ip)

  Stack.exports(
    nginxHostname = p"http://${hostnameOrIp.getOrElse("unknown")}:$nginxPortNumber"
  )
```

### PR DESCRIPTION
**Pull request by pawelprazak**
Fixes #424

### CODE REVIEW

**Comment by pawelprazak** [comment]
Added the requested `getOrFail`, and scaladocs for public methods.

File: core/src/main/scala/besom/internal/Output.scala [hunk 1]
```diff
@@ -113,15 +244,162 @@ trait OutputExtensionsFactory:
...
+      */
+    def headOption: Output[Option[A]] = output.map(_.headOption)
+
+    /** Calls [[List.lastOption]] on the underlying [[List]]
+      * @return
+      *   an [[Output]] of [[Option]] of the last element of the list
+      */
+    def lastOption: Output[Option[A]] = output.map(_.lastOption)
+
+    /** Calls [[List.tail]] on the underlying [[List]], but does not throw on `Nil`
```
Comment (lbialy) [inline core/src/main/scala/besom/internal/Output.scala:317]: fail instead of throw

File: core/src/main/scala/besom/internal/Output.scala [hunk 2]
```diff
@@ -113,15 +244,162 @@ trait OutputExtensionsFactory:
...
+      * @return
+      *   an [[Output]] of the `tail` of the [[List]] or an empty list if the list is empty
+      */
+    def tailOrEmpty: Output[List[A]] = output.map {
+      case Nil  => Nil
+      case list => list.tail
+    }
+
+    /** Calls [[List.init]] on the underlying [[List]], but does not throw on `Nil`
```
Comment (lbialy) [inline core/src/main/scala/besom/internal/Output.scala:326]: fail instead of throw

**Review by lbialy — CHANGES_REQUESTED** [review]

**Comment by pawelprazak** [comment]
Addressed the comments. Extended signatures of mapInner / flatMap inner for lists to match option. Added unit tests.

**Review by lbialy — APPROVED** [review]

**Pull request merged**